src/draw/sw/lv_draw_sw_img.c

```diff
@@ -236,22 +236,35 @@ static void img_draw_core(lv_draw_unit_t * draw_unit, const lv_draw_image_dsc_t
     }
     /*Handle masked RGB565, RGB888, XRGB888, or ARGB8888 images*/
     else if(!transformed && masked && draw_dsc->recolor_opa <= LV_OPA_MIN) {
+        lv_image_decoder_dsc_t mask_decoder_dsc;
+        lv_area_t mask_area;
+        lv_result_t decoder_res = lv_image_decoder_open(&mask_decoder_dsc, draw_dsc->bitmap_mask_src, NULL);
+        if(decoder_res == LV_RESULT_OK && mask_decoder_dsc.decoded) {
+            const lv_draw_buf_t * mask_img = mask_decoder_dsc.decoded;
+            blend_dsc.mask_buf = mask_img->data;
```

Comment:
Let's check if the image color format is A8 and ignore it here.

In future, we can add decoder args "as_mask" etc. to let decoder convert image to A8 format.

Reply:
Fair point, added.